src/components/page-header/image-panel-modal.tsx

```diff
@@ -0,0 +1,454 @@
+                const reader = new FileReader();
+                reader.onload = () => {
+                    if (typeof reader.result === 'string') {
+                        const imgId = `l${nanoid(5)}`; // Use timestamp to ensure unique ID
+                        imageStoreIndexedDB.save(imgId, reader.result);
+                        setLocalList(prev => [...prev, { id: imgId, thumbnail: reader.result as string }]);
+                        setRefresh();
+                    }
+                };
+                reader.readAsDataURL(file);
+            }
+        }
+    };
+
+    async function fetchImageAsBase64(url: string): Promise<string> {
+        const response = await fetch(url, {
+            headers: {
+                Authorization: `Bearer ${token}`,
+            },
+        });
+        if (!response.ok) {
+            throw new Error(`Failed to download: ${response.statusText}`);
+        }
+
+        const blob = await response.blob();
+        return await blobToBase64(blob);
+    }
+
+    function blobToBase64(blob: Blob): Promise<string> {
+        return new Promise((resolve, reject) => {
+            const reader = new FileReader();
+            reader.onloadend = () => {
+                if (typeof reader.result === 'string') {
+                    resolve(reader.result);
+                } else {
+                    reject('Failed to convert blob to base64');
+                }
+            };
+            reader.onerror = reject;
+            reader.readAsDataURL(blob);
+        });
+    }
+
+    const fetchImageList = async () => {
+        if (!token) return;
+        const rep = await fetch(image_endpoint, {
+            headers: {
+                accept: 'application/json',
+                'Content-Type': 'application/json',
+                Authorization: `Bearer ${token}`,
+            },
+        });
+        if (rep.status !== 200) {
+            return;
+        }
+        const { images } = await rep.json();
+
+        const list = await Promise.all(
+            images.map(async (img: { id: string; hash: string }) => {
+                const thumbnail = await fetchImageAsBase64(`${image_endpoint}/thumbnail/${img.id}`);
+                return {
+                    id: `s${img.id}`,
+                    hash: img.hash,
+                    thumbnail,
+                };
+            })
+        );
+        setCloudList(list);
+    };
+
+    const getLocalImageList = async () => {
+        const list: ImageList[] = [];
+        (await imageStoreIndexedDB.getAll()).forEach((img, id) => {
+            if (img && id.startsWith('l')) {
+                list.push({
+                    id,
+                    thumbnail: img,
+                });
+            }
+        });
+        setLocalList(list);
+    };
+
+    React.useEffect(() => {
+        if (isOpen) {
+            setLoading(true);
+            setError(undefined);
+            setSelected(undefined);
+            getLocalImageList();
```

Comment:
Should we await to make sure loading is set to false when this runs to complete?